Implement LCD touch calibration

EvanvanUrk opened the pull request:
This implements calibration for the MI0283QT touchscreen based on the code in Watterott's arduino-lib repository.

EvanvanUrk committed:
Implement LCD touch calibration

jorams commented on src/pleasant-lcd.h:
```diff
@@ -145,6 +145,8 @@ enum lcd_orientation {
 };
 
+enum lcd_orientation lcd_current_orientation;
+
```
Move this to the State block below, and make it `extern` like everything there. Then initialize it to a sane value at the top of `pleasant-lcd.c`

jorams commented on src/pleasant-lcd.h:
```diff
@@ -234,6 +236,50 @@ void lcd_fill_rect(uint16_t x,
                    lcd_color color);
 
+/* Touch ------------------------------------------------------------------- */
+typedef struct {
```
Please keep one blank line below the section heading.

jorams commented on src/pleasant-lcd.h:
```diff
@@ -234,6 +236,50 @@ void lcd_fill_rect(uint16_t x,
+typedef struct {
+  uint32_t x;
+  uint32_t y;
+} CAL_POINT;
```
I'd prefer it if everything exported from this library started with `lcd_` and if structs were not `typedef`'d away. Like this:

```c
struct lcd_calibration_point {
  uint32_t x;
  uint32_t y;
};
```

jorams commented on src/pleasant-lcd.h:
```diff
@@ -234,6 +236,50 @@ void lcd_fill_rect(uint16_t x,
+  uint_least32_t div;
+} CAL_MATRIX;
+
+CAL_MATRIX tp_matrix;
```
Move this to the State block below, and make it `extern` like everything there. Then initialize it to a sane value at the top of `pleasant-lcd.c`

jorams commented on src/pleasant-lcd.c:
```diff
@@ -300,6 +304,107 @@ void lcd_fill_rect(uint16_t x,
+                  ((tp[1].x - tp[2].x) * (tp[0].y - tp[2].y));
+
+  if(tp_matrix.div == 0) {
+    return; /* error */
```
Would be nice if this error was propagated somewhere.

jorams commented on src/pleasant-lcd.c:
```diff
@@ -375,3 +480,95 @@ bool lcd_touch_read(uint16_t *touch_pressure,
+    y = tp_y;
+    x = ((tp_matrix.a * x) + (tp_matrix.b * y) + tp_matrix.c) / tp_matrix.div;
+
+    switch(lcd_current_orientation) {
```
Put a space after the `switch` and after all the `if`s below.

jorams commented on src/pleasant-lcd.c:
```diff
@@ -375,3 +480,95 @@ bool lcd_touch_read(uint16_t *touch_pressure,
+    y = tp_y;
+    y = ((tp_matrix.d * x) + (tp_matrix.e * y) + tp_matrix.f) / tp_matrix.div;
+
+    switch(lcd_current_orientation) {
```
Put a space after the `switch` and after all the `if`s below.

jorams reviewed (changes requested):
Mostly aesthetic points of improvement, with one general guideline: Make sure everything exported has an `lcd_` or `LCD_` prefix.

EvanvanUrk commented:
Fixed formatting issues, removed the return upon error in `lcd_touch_set_calibration` since there really isn't a way to determine what caused the error in the first place and moved the structs and definitions over to the .c file because I don't feel they need to be exposed in the first place now that I look at it.
